shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -40,6 +40,12 @@
   private static final String ENABLE_SKPARAGRAPH_META_DATA_KEY =
       "io.flutter.embedding.android.EnableSkParagraph";
 
+  /**
+   * Set whether leave or clean up the VM after the last shell shuts down. It's true by default, set
+   * it to false to destroy VM.
```

Comment:
can you add an example use case, or more context about why someone may or may not want to leak the VM? 
e.g. is this problematic when using shared engines and/or add-to-app with some other set up?

Reply:
> can you add an example use case, or more context about why someone may or may not want to leak the VM? e.g. is this problematic when using shared engines and/or add-to-app with some other set up?

Can we also add more context with the comment in `settings.h` into this java and objc docs:
in `engine/common/settings.h`
```c++
 // All shells in the process share the same VM. The last shell to shutdown 
 // should typically shut down the VM as well. However, applications depend on 
 // the behavior of "warming-up" the VM by creating a shell that does not do 
 // anything. This used to work earlier when the VM could not be shut down (and 
 // hence never was). Shutting down the VM now breaks such assumptions in 
 // existing embedders. To keep this behavior consistent and allow existing 
 // embedders the chance to migrate, this flag defaults to true. Any shell 
 // launched with this flag set to true will leak the VM in the process. There 
 // is no way to shut down the VM once such a shell has been started. All 
 // shells in the platform (via their embedding APIs) should cooperate to make 
 // sure this flag is never set if they want the VM to shutdown and free all 
 // associated resources. 
```

Reply:
DONE~ Add more comments, leave a TODO and added a sample code:
* https://github.com/flutter/samples/pull/996/